sql/2025/04/slider-usage.sql

```diff
@@ -0,0 +1,72 @@
+  # HTTP Archive query to get % of WordPress sites using a slider/carousel library.
+  #
+  # WPP Research, Copyright 2025 Google LLC
+  #
+  # Licensed under the Apache License, Version 2.0 (the "License");
+  # you may not use this file except in compliance with the License.
+  # You may obtain a copy of the License at
+  #
+  #     https://www.apache.org/licenses/LICENSE-2.0
+  #
+  # Unless required by applicable law or agreed to in writing, software
+  # distributed under the License is distributed on an "AS IS" BASIS,
+  # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  # See the License for the specific language governing permissions and
+  # limitations under the License.
+
+DECLARE
+  DATE_TO_QUERY DATE DEFAULT '2025-03-01';
+
+CREATE TEMPORARY FUNCTION
+  HAS_TECHNOLOGY(technologies ARRAY<STRUCT<technology STRING,
+    categories ARRAY<STRING>,
+    info ARRAY<STRING>>>,
+    technologyToFind STRING)
+  RETURNS BOOL AS ( EXISTS(
+    SELECT
+      *
+    FROM
+      UNNEST(technologies) AS technology
+    WHERE
+      technology.technology = technologyToFind ) );
+
+CREATE TEMPORARY FUNCTION
+  HAS_SLIDER(technologies ARRAY<STRUCT<technology STRING,
+    categories ARRAY<STRING>,
+    info ARRAY<STRING>>>)
+  RETURNS BOOL AS (
+    EXISTS(
+    SELECT
+      *
+    FROM
+      UNNEST(technologies) AS technology
+    WHERE
+      technology.technology IN (
+        'FlexSlider',
+        'Flickity',
+        'Master Slider',
+        'MetaSlider',
+        'OWL Carousel',
+        'Slick',
+        'Slider Revolution',
+        'Smart Slider 3',
+        'Swiper'
+      )
+    )
+  );
```

Comment:
This is pretty much the same as `HAS_TECHNOLOGY`, except that it looks for one of multiple technologies.

You could get rid of the two functions in favor of a tiny change to make `HAS_TECHNOLOGY` expect an array of strings instead of a single string. Then you can pass all the technologies to the function.

You could keep `HAS_SLIDER()` if you think it has value but have it call `HAS_TECHNOLOGIES`.

Not a big deal, but avoids duplicate implementation of almost the same piece of query.

Reply:
Yeah I thought about that but figured it doesn't really matter for a one off query.